### DISCUSSION TITLE
React Hook "useContractRead" is called conditionally. React Hooks must be called in the exact same order in every component render

### DISCUSSION THREAD

**Discussion by dobbobalina2** [wevm/wagmi#560] [Q&A]
I want to bring this thread up again becausae the current answer for this issue is to set `enable: false`.

But Currently that doesn't actually fix the issue. Heres what I have

 if (account != undefined) {
    const { data: balance} = useContractRead(
      {
        addressOrName: mintingAddress,
        contractInterface: mintingInterface,
      },
      "balanceOf",
      {
        onError: (error) => {
          if (error.message.includes("interface")) {
            console.log(error);
          } else {
            console.log(error.message);
            toast.error(error.message);
          }
        },
        chainId: chain.rinkeby.id,

        enabled: false,
        args: [account?.address],
        overrides: { from: account?.address },
      }
    );
}

**Comment by jxom** [comment]
Hooks cannot be called in conditionals: https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

If you only want to invoke the contract read when an account exists, please make use of the `enabled` attribute:

```tsx
useContractRead({
  ...
  enabled: account?.address
  ...
})
```

**Reply by dobbobalina2** [reply]
OHHHH THAT's how that works?

That should be made more clear that I can pass stuff like that in. 

I get a typescript error when I do about passing strings to a  type boolean.  so instead I did this

```
useContractRead({
  ...
  enabled: !!account?.address
  ...
})

```

That has  fixed most of the issues I get but I still get  a race condition `" contract.interface.encodeFunctionData is not a function"`

**Reply by dobbobalina2** [reply]
Found the fix.

```
useContractRead({
  ...
  enabled: !account?.address
  ...
})

```

That worked. I'm guessing because you want it to be trying to update aka true, as long as the account isn't connected. idk it works

**Reply by jasan-s** [reply]
This should be better documented

**Reply by tmm** [reply]
Let us know if you have ideas on how to document it better! `enabled` is a standard option across all query hooks in wagmi and is in the docs (e.g. `useContractRead` [`enabled`](https://wagmi.sh/docs/hooks/useContractRead#enabled-optional)).
